.github/workflows/ci.yml

```diff
@@ -1,17 +1,20 @@
-name: ci
-on: push
+name: CI
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, edited, synchronize, reopened]
```

Comment:
It could, that depends on how we want to tackle it. IMO, if someone commits it doesn't mean necessarily that the CI should run. When someone open a PR tho, it means the changes are 'cooked' and ready for CI and review.

That's my two cents tho, we can put it on push: branches: *